### DISCUSSION TITLE
stochastictool - transient_outputs

### DISCUSSION THREAD

**Discussion by purepuremilk-zz** [idaholab/moose#18139] [Q&A Modules: General] [answered]
Hello, I am running a transient problem with stochastictool. In my transient input file, I have a vectorprocessor that dumps vectors at every time step, which gives me 100 CSVs if I am having 100 steps and output as CSVs. My postprocessor looks like: 

`[VectorPostprocessors]
  [T_sample]
    type = LineValueSampler
    num_points = 100
    start_point = '0 0.5 -2.5'
    end_point =   '0 0.5 2.5'
    variable = T
    sort_by = z
  []
[]`

Now I am using stochastictool to sample the boundary conditions and would like to gather outputs all transient data from each sample. I am currently using 'Reporter' and 'Transfers'. Meanwhile, I used JSON output as instructed in issue #17895. However, I am having only one vector for each sample instead of all 100 steps in the JSON file. What should I do to output all 100 steps for each sample? Follows are my code for 'Reporter', 'Transfers', and 'Outputs'

`[Transfers]
  [data]
    type = SamplerReporterTransfer
    multi_app = runner
    execute_on = 'timestep_end'
    sampler = hypercube
    stochastic_reporter = storage
    from_reporter = 'T_sample/z T_sample/T'
  []
[]
[Reporters]
  [storage]
    type = StochasticReporter
    execute_on = 'timestep_end'
  []
[]
[Outputs]
  [out]
    type = JSON
    execute_system_information_on = none
  []
[]`

**Comment by zachmprince** [comment, marked as answer]
If you are using SamplerFullSolveMultiApp, this capability is not available, although we are actively working on it. Another option is to use SamplerTransientMultiApp, but you will need to have a constant time step defined in the main input. This will perform the transfer and output at each time step.

**Reply by aeslaughter** [reply]
This will also cause a significant slow down because the simulations need to stop at every time step. I think we will be doing some work in this area soon(ish).
